### PR TITLE
Bugfix/Rotation precision fix

### PR DESCRIPTION
**Pull request by darek607**
Fixed the precision of rotation extracted from transformation matrix (in ArCore we cannot assign a transform matrix directly - the only solution is to extract translation, scale and rotation and assign them separately to the Node - this require a very high precision when transform matrix contains both scale and rotation).

Transformations scene before fix (rotations do not match)

rot_before

After fix

rotation_fix

### CODE REVIEW

File: android/src/main/java/com/magicleap/magicscript/utils/Extensions.kt [hunk 1]
```diff
@@ -96,6 +97,58 @@ fun Vector3.rotatedBy(quaternion: Quaternion): Vector3 {
...
+ * It removes the scale with a high precision using BigDecimal class.
+ *
+ * It should be used before extracting a rotation when the matrix contains
+ * also a scale transformation, else the standard decompose functions
+ * (decomposeRotation or extractRotation) will return a very low precision quaternion.
+ *
+ * @param decomposedScale previously decomposed scale from the matrix
+ *
+ * Based on com.google.ar.sceneform.math.Matrix.java
+ */
```
Comment (pleszkiewicz) [inline android/src/main/java/com/magicleap/magicscript/utils/Extensions.kt:115]: Code looks ok, but it's really strange that you need to switch to BigDecimal (Double?) to calculate/extract anything from the Matrix.

Comment (pleszkiewicz) [reply android/src/main/java/com/magicleap/magicscript/utils/Extensions.kt:115]: Have you found any info about this issue in the Internet?

Comment (darek607) [reply android/src/main/java/com/magicleap/magicscript/utils/Extensions.kt:115]: Yes, however not directly related to ArCore. From [this](https://computergraphics.stackexchange.com/questions/4281/is-it-possible-to-turn-a-3d-rotation-matrix-4x4-into-its-component-parts-rota) :

Comment (darek607) [reply android/src/main/java/com/magicleap/magicscript/utils/Extensions.kt:115]: I also carefully checked the ARCore's decomposition algorithm and compared with others - it's correct. Yes, I'm also surprised, but in this case precision really matter :-)

**Pull request merged**